File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int count = 0;
        int i = 0;

        while (i < nums.size() - count) {
            if (nums[i] == val) {
                for (int j = i; j < nums.size() - 1; j++) {
                    int dummy = nums[j+1];
                    nums[j+1] = val;
                    nums[j] = dummy;
                }
                count++;
            } else {
                i++;
            }

            cout << "[";
            for (int num : nums) {
                cout << num << ",";
            }
            cout << "]";
            cout << "\n";
        }
        
        return nums.size() - count;
    }
};

File: Data Structures & Algorithms/special-array-with-x-elements-greater-than-or-equal-x/submission-1.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        count = [0 for i in range(1001)]

        for num in nums:
            count[num] += 1

        i = len(count) - 2

        while i >= 0:
            count[i] += count[i+1]
            i-=1


        for i, x in enumerate(count):
            if i == x:
                return x


        return -1

        
        